### PR TITLE
Add support for PHP 8.4

### PR DESCRIPTION
**Pull request by cedric-anne**
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | no
| New Feature   | yes
| RFC           | no
| QA            | no

### CODE REVIEW

**Comment by cedric-anne** [comment]
There are the following deprecations triggered by any PHP command in the CI env itself:
```
PHP Deprecated:  session.sid_length INI setting is deprecated in Unknown on line 0
PHP Deprecated:  session.sid_bits_per_character INI setting is deprecated in Unknown on line 0
```

For instance, they are triggered inside the CI init process itself:
image

I do not know how to fix them, and it makes the PHPUnit test failing.

**Comment by gsteel** [comment]
An upgrade to PHPUnit 10 is probably required - see #190

**Comment by cedric-anne** [comment]
> An upgrade to PHPUnit 10 is probably required - see #190

Honestly, I thought it would take me only a few minutes at first. I proposed this to be able to activate PHP 8.4 checks on the CI of a dependency of a dependency we use. Unfortunately, I will not have time to investigate more on this right now.

File: composer.json [hunk 1]
```diff
@@ -34,7 +34,7 @@
         }
     },
     "require": {
-        "php": "~8.1.0 || ~8.2.0 || ~8.3.0",
+        "php": "~8.1.0 || ~8.2.0 || ~8.3.0 || ~8.4.0",
         "psr/http-factory": "^1.0.2",
```
Comment (hafezdivandari) [inline composer.json]:
```suggestion
        "psr/http-factory": "^1.1",
```

**Comment by cedric-anne** [comment]
I have no idea how to fix tests and no time to investigate on this. I close this PR as I am pretty sure I will not be able to finish it.

**Comment by hafezdivandari** [comment]
Hi @froschdesign, any plan for this?

**Pull request closed without merging**